File: assuit_sheets/sheet_2/I/I.js
// Function to check if a number is a palindrome
function isPalindrome(number) {
    const numStr = number.toString();
    const reversedStr = numStr.split('').reverse().join('');
    return numStr === reversedStr;
}

// Function to remove leading zeroes
function removeLeadingZeroes(number) {
    const numStr = number.toString();
    const trimmedStr = numStr.replace(/^0+/, '');
    return parseInt(trimmedStr, 10);
}

// Example usage
const N = 1203400;

// Reversed number without leading zeroes
const reversedNumber = removeLeadingZeroes(parseInt(N.toString().split('').reverse().join('')));

// Check if it's a palindrome
if (isPalindrome(N)) {
    console.log("YES");
} else {
    console.log("NO");
}

// Print the reversed number without leading zeroes
console.log(reversedNumber);

File: assuit_sheets/sheet_2/I/I.cpp
#include <iostream>
#include <string>
using namespace std;

bool isPalindrome(string s) {
    int n = s.length();
    for (int i = 0; i < n / 2; i++) {
        if (s[i] != s[n - i - 1]) {
            return false;
        }
    }
    return true;
}

int main() {
    string s;
  
    cin >> s;

    string reversed;
    bool leadingZero = true;
    for (int i = s.length() - 1; i >= 0; i--) {
        if (s[i] != '0') {
            leadingZero = false;
        }
        if (!leadingZero) {
            reversed.push_back(s[i]);
        }
    }

    cout << reversed << endl;
    if (isPalindrome(s)) {
        cout << "YES" << endl;
    } else {
        cout << "NO" << endl;
    }

    return 0;
}
